library/sha256.c

```diff
@@ -57,11 +57,26 @@
 #include "mbedtls/platform.h"
 
 #if defined(__aarch64__)
+
 #  if defined(MBEDTLS_SHA256_USE_A64_CRYPTO_IF_PRESENT) || \
     defined(MBEDTLS_SHA256_USE_A64_CRYPTO_ONLY)
+
 /* *INDENT-OFF* */
+
+#   ifdef __ARM_NEON
+#       include <arm_neon.h>
+#   else
+#       error "Target does not support NEON instructions"
```

Comment:
I don't think the error here is correct. If the user has defined `MBEDTLS_SHA256_USE_A64_CRYPTO_IF_PRESENT` then we should include and use `arm_neon`, if it is available, and fall back to plain C if not.

Reply:
I am not sure if I understand this comment. 

`__ARM_NEON` is a build time feature detection. If it is not available, compiler will report `arm_neon.h not available`.  This error is just report the real reason.

Reply:
On closer reading of `mbedtls_config`, it says: If MBEDTLS_SHA256_USE_A64_CRYPTO_IF_PRESENT is defined when building for a non-Aarch64 build it will be silently ignored.

In this case I guess we have an Aarch64 build, but can't use runtime detection. It would be surprising to the user if the build works but there is no runtime detection. So actually I think it is correct as it is.